Runar/Runar/Services/Memory/CoreData/CoreDataManager.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  CoreDataManager.swift
+//  Runar
+//
+//  Created by Alexey Poletaev on 23.09.2022.
+//
+
+import Foundation
+import CoreData
+import UIKit
+
+final class CoreDataManager {
+
+    // MARK: - Properties
+    static let shared = CoreDataManager()
+    private let nameCoreData = "CoreData"
+    var generatorIsLoaded: Bool = false
+    var libraryIsLoaded: Bool = false
+    private var context = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+    private lazy var persistentContainer: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: nameCoreData)
+        container.loadPersistentStores { _, error in
+            if let error = error {
+                debugPrint("Unresolved error \(error)")
+            }
+        }
+        return container
+    }()
+    private var parentContext: NSManagedObjectContext {
```

Comment:
Why for?

Reply:
Parent context is required to access the Core Data persistent store